src/local_scheduler/local_scheduler_extension.cc

```diff
@@ -111,14 +119,22 @@ static PyObject *PyLocalSchedulerClient_compute_put_id(PyObject *self,
                                                        PyObject *args) {
   int put_index;
   TaskID task_id;
-  if (!PyArg_ParseTuple(args, "O&i", &PyObjectToUniqueID, &task_id,
-                        &put_index)) {
+  PyObject *use_raylet;
+  if (!PyArg_ParseTuple(args, "O&iO", &PyObjectToUniqueID, &task_id, &put_index,
+                        &use_raylet)) {
     return NULL;
   }
-  ObjectID put_id = task_compute_put_id(task_id, put_index);
-  local_scheduler_put_object(
-      ((PyLocalSchedulerClient *) self)->local_scheduler_connection, task_id,
-      put_id);
+  ObjectID put_id;
+  if (!PyObject_IsTrue(use_raylet)) {
+    put_id = task_compute_put_id(task_id, put_index);
+    local_scheduler_put_object(
+        ((PyLocalSchedulerClient *) self)->local_scheduler_connection, task_id,
+        put_id);
+  } else {
+    // TODO(rkn): Raise an exception if the put index is not a valid value
```

Comment:
Is this in case we reach the maximum number of puts for a task? Otherwise, this should be a bug, right?

Reply:
It would indeed be a bug, but it should be a bug that leads to a Python exception instead of a segfault. In general, when writing Python code, incorrect arguments passed into functions should not cause crashes.